labman/db/plate.py

```diff
@@ -300,3 +390,59 @@ def get_well(self, row, column):
                     % (row, column, self.id))
 
             return container_module.Well(res[0][0])
+
+    def get_wells_by_sample(self, sample_id):
+        """Returns the list of wells containing the given sample
+
+        Parameters
+        ----------
+        sample_id : str
+            The sample to search for
+
+        Returns
+        -------
+        list of labman.db.container.Well
+        """
+        with sql_connection.TRN as TRN:
+            sql = """SELECT well_id
+                     FROM qiita.well
+                        JOIN qiita.composition USING (container_id)
+                        JOIN qiita.sample_composition USING (composition_id)
+                     WHERE plate_id = %s AND sample_id = %s
+                     ORDER BY well_id"""
+            TRN.add(sql, [self.id, sample_id])
+            res = [container_module.Well(well)
```

Comment:
Same here, I think it not only will be faster but also cleaner.

Reply:
Same as above - it is just constructing the object.

Reply:
K, thanks.